lib/matplotlib/cm.py

```diff
@@ -356,13 +356,15 @@ def to_rgba(self, x, alpha=None, bytes=False, norm=True):
                     xx[:, :, :3] = x
                     xx[:, :, 3] = alpha
                 elif x.shape[2] == 4:
-                    xx = x
+                    xx = x.copy()
                 else:
                     raise ValueError("Third dimension must be 3 or 4")
                 if xx.dtype.kind == 'f':
                     if norm and (xx.max() > 1 or xx.min() < 0):
                         raise ValueError("Floating point image RGB values "
                                          "must be in the 0..1 range.")
+                    # If any of R, G, B, or A is nan, set to 0
+                    xx[np.any(np.isnan(x), axis=2), :] = 0
```

Comment:
The casting for the `bytes=True` case will error if there are nans in the array so we need to replace the nan with an arbitrary float.  Setting the whole pixel to 0 seems the simplest way to do that.

Reply:
I think this change is too late; if `xx = np.array([10, np.nan, 0.2])`, then the above check should fail, but it doesn't, because `xx.max() == np.nan`, and `np.nan > 1` is False.

Reply:
If I just move this above the range check, then we replace the pixel with zeros which will make the range check pass.  So there will be no difference.  Do we want to do something more complicated than set the whole pixel to zero?

Reply:
That depends on if we want to ignore all of the pixel that is NaN, regardless of value or not. We can replace with `np.nanmax`/`np.nanmin` in that case. If not, then we need to do something more complicated.

But my point is more that the check is broken even if the NaN and the invalid value are not in the same pixel.

Reply:
> But my point is more that the check is broken even if the NaN and the invalid value are not in the same pixel.

Oh yes, I get it now.